src/Shared/JsonSchemaExporter/JsonSchemaExporter.cs

```diff
@@ -0,0 +1,1128 @@
+        [typeof(Int128)] = numberHandling => GetSchemaForNumericType(JsonSchemaType.Integer, numberHandling),
+#endif
+        [typeof(char)] = _ => new JsonSchema { Type = JsonSchemaType.String, MinLength = 1, MaxLength = 1 },
+        [typeof(string)] = _ => new JsonSchema { Type = JsonSchemaType.String },
+        [typeof(byte[])] = _ => new JsonSchema { Type = JsonSchemaType.String },
+        [typeof(Memory<byte>)] = _ => new JsonSchema { Type = JsonSchemaType.String },
+        [typeof(ReadOnlyMemory<byte>)] = _ => new JsonSchema { Type = JsonSchemaType.String },
+        [typeof(DateTime)] = _ => new JsonSchema { Type = JsonSchemaType.String, Format = "date-time" },
+        [typeof(DateTimeOffset)] = _ => new JsonSchema { Type = JsonSchemaType.String, Format = "date-time" },
+        [typeof(TimeSpan)] = _ => new JsonSchema
+        {
+            Comment = "Represents a System.TimeSpan value.",
+            Type = JsonSchemaType.String,
+            Pattern = @"^-?(\d+\.)?\d{2}:\d{2}:\d{2}(\.\d{1,7})?$",
+        },
+
+#if NET6_0_OR_GREATER
+        [typeof(DateOnly)] = _ => new JsonSchema { Type = JsonSchemaType.String, Format = "date" },
+        [typeof(TimeOnly)] = _ => new JsonSchema { Type = JsonSchemaType.String, Format = "time" },
+#endif
+        [typeof(Guid)] = _ => new JsonSchema { Type = JsonSchemaType.String, Format = "uuid" },
+        [typeof(Uri)] = _ => new JsonSchema { Type = JsonSchemaType.String, Format = "uri" },
+        [typeof(Version)] = _ => new JsonSchema
+        {
+            Comment = "Represents a version string.",
+            Type = JsonSchemaType.String,
+            Pattern = @"^\d+(\.\d+){1,3}$",
+        },
+
+        [typeof(JsonDocument)] = _ => JsonSchema.True,
+        [typeof(JsonElement)] = _ => JsonSchema.True,
+        [typeof(JsonNode)] = _ => JsonSchema.True,
+        [typeof(JsonValue)] = _ => JsonSchema.True,
+        [typeof(JsonObject)] = _ => new JsonSchema { Type = JsonSchemaType.Object },
+        [typeof(JsonArray)] = _ => new JsonSchema { Type = JsonSchemaType.Array },
+    };
+
+    // Adapted from https://github.com/dotnet/runtime/blob/release/9.0/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/JsonPrimitiveConverter.cs#L36-L69
+    private static JsonSchema GetSchemaForNumericType(JsonSchemaType schemaType, JsonNumberHandling numberHandling, bool isIeeeFloatingPoint = false)
+    {
+        Debug.Assert(schemaType is JsonSchemaType.Integer or JsonSchemaType.Number, "schema type must be number or integer");
+        Debug.Assert(!isIeeeFloatingPoint || schemaType is JsonSchemaType.Number, "If specifying IEEE the schema type must be number");
+
+        string? pattern = null;
+
+        if ((numberHandling & (JsonNumberHandling.AllowReadingFromString | JsonNumberHandling.WriteAsString)) != 0)
+        {
+            pattern = schemaType is JsonSchemaType.Integer
+                ? @"^-?(?:0|[1-9]\d*)$"
```

Comment:
These are patterns that show up only in the schema, right? We don't actually execute this in .NET code? (If we did, we could improve / specialize them a bit.)

Reply:
No execution. This is expected to be executed by a validation component like the one present in JsonSchema.NET.